src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -254,7 +251,7 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
 
             if (!modifierErrors)
             {
-                this.CheckModifiers(location, hasBody || hasExpressionBody, isAutoPropertyAccessor, diagnostics);
+                this.CheckModifiers(location, hasBody || hasExpressionBody, _accessorBodyShouldBeSynthesized, diagnostics);
```

Comment:
It looks like this can be folded into the previous parameter value and this parameter can be removed.